taf/testlib/linux/collectd.py

```diff
@@ -129,21 +137,25 @@ def command_generator(action, plugin, collectd_conf):
     """
     def method(**params):
         command_list = []
-        if ACTIONS[action]['kwargs_required'] and not params:
+        this_action = ACTIONS[action]
+        this_cmd = this_action['cmd']
+        kwargs_needed = this_action['kwargs_required']
```

Comment:
It seems odd that the key string is different that the string of the name its value gets assigned to.  I'd expect:

```kwargs_required = this_action['kwargs_required']```

It also occurs to me that this could have a default value:

```kwargs_required = this_action.get('kwargs_required', False)```